test/unit/lib/request-with-oauth2.js

```diff
@@ -53,72 +53,58 @@ describe('lib/requestWithOAuth2', function() {
 
-    it('Uses an existing authorization token if present', function(done) {
-      // First call to oAuth server
-      requestPost.yields(null, { statusCode: 200 }, JSON.stringify(successfulAuthResponse));
-      // First call to URL
-      requestGet.yields(null, { statusCode: 200 }, 'Woo');
-      requestWithOAuth2.get('http://testhost.com', 'http://oauthserver.com',
-        'clientId', 'clientSecret', 'username', 'password', function(err) {
-        if (!err) {
-          request.post.should.have.been.calledWith(expectedOAuthRequest);
-          request.get.should.have.been.calledWith(expectedTesthostRequest);
-          requestWithOAuth2.get('http://testhost.com', 'http://oauthserver.com', 'clientId',
-            'clientSecret', 'username', 'password', function(err2) {
-            // Check it doesn't authorize with multiple requests
-            if (!err2) {
-              assert(request.post.calledOnce);
-            }
-            done(err2);
-          });
-        } else {
-          done(err);
-        }
+    it('Uses an existing authorization token if present', function (done) {
+      //First call to oAuth server
+      requestPost.yields(null, {statusCode: 200}, JSON.stringify(successfulAuthResponse));
+      //First call to URL
+      requestGet.yields(null, {statusCode: 200}, 'Woo');
+      requestWithOAuth2.get('http://testhost.com', 'http://oauthserver.com', 'clientId', 'clientSecret', 'username', 'password', function (err, res, body) {
+        request.post.should.have.been.calledWith(expectedOAuthRequest);
+        request.get.should.have.been.calledWith(expectedTesthostRequest);
+        requestWithOAuth2.get('http://testhost.com', 'http://oauthserver.com', 'clientId', 'clientSecret', 'username', 'password', function (err, res, body) {
+          //Check it doesn't authorize with multiple requests
+          assert(request.post.calledOnce);
+          done();
+        });
       });
     });
 
-    it('Will return the error from the oAuth request if there is one', function(done) {
-      // First call to oAuth server
-      requestPost.yields(null, { statusCode: 400 }, {});
-      // First call to URL
-      requestWithOAuth2.get('http://testhost.com', 'http://oauthserver.com', 'clientId',
-        'clientSecret', 'username', 'password', function(err, res) {
-        err.should.be.an('Error');
+    it('Will return the error from the oAuth request if there is one', function (done) {
+      //First call to oAuth server
+      requestPost.yields(null, {statusCode: 400}, {});
+      //First call to URL
+      requestWithOAuth2.get('http://testhost.com', 'http://oauthserver.com', 'clientId', 'clientSecret', 'username', 'password', function (err, res, body) {
         request.post.should.have.been.calledWith(expectedOAuthRequest);
         sinon.assert.notCalled(request.get);
-        assert(res.statusCode === 400, 'Status code not 400 as expected');
+        assert(res.statusCode == 400, 'Status code not 400 as expected');
```

Comment:
Why not using === ? Thought that was generally safer as you don't get funny results if some weird type conversion happens